### PR TITLE
pkg/asset: Update kube-dns to v17.1

### LINKED ISSUE #93: Use latest kube-dns
kube-dns got a rewrite in 1.3, and is required for features like federation to work properly. It's also lighter weight/simpler. Bootkube is deploying the old kube-dns, it should get upgraded (also while at it the file should be renamed to a -deployment suffix since its no longer an RC).

### PR DESCRIPTION
**Pull request by chancez**
Closes #93

### CODE REVIEW

**Comment by aaronlevy** [comment]
Thank you for updating this!

Can we have it track the v1.3 release version (rather than head)? Looks like that's still on v17.1:
https://github.com/kubernetes/kubernetes/blob/v1.3.5/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base#L34

File: pkg/asset/templates/kube-dns-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            path: /readiness
+            port: 8081
+            scheme: HTTP
+          # we poll on pod startup for the Kubernetes master service and
+          # only setup the /readiness HTTP server once that's available.
+          initialDelaySeconds: 30
+          timeoutSeconds: 5
+        args:
+        # command = "/kube-dns"
+        - --domain=cluster.local
```
Comment (aaronlevy) [inline pkg/asset/templates/kube-dns-deployment.yaml]: Also we should keep the trailing "." here (easy to miss): https://github.com/kubernetes/kubernetes/blob/v1.3.5/cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base#L76

**Comment by chancez** [comment]
@aaronlevy switched it over to v17.1

**Comment by aaronlevy** [comment]
@chancez mind updating this again? templating has changed...

**Comment by dghubble** [comment]
He's on vacation. We're seeing flakiness in v11 and would like to update to v17.1. #126

**Comment by aaronlevy** [comment]
fixed in: https://github.com/kubernetes-incubator/bootkube/pull/126

**Pull request closed without merging**